client/src/app/site/base/projectable.ts

```diff
@@ -1,32 +1,34 @@
-import { ProjectorOptions } from './projector-options';
 import { Displayable } from 'app/shared/models/base/displayable';
+import { IdentifiableProjectorElement } from 'app/shared/models/core/projector';
+import { SlideOptions } from './slide-options';
 
-/**
- * Interface for every model, that should be projectable.
- */
-export interface Projectable extends Displayable {
-    /**
-     * All options for the slide
-     */
-    getProjectorOptions(): ProjectorOptions;
+export function isProjectorElementBuildDeskriptor(obj: any): obj is ProjectorElementBuildDeskriptor {
+    const deskriptor = <ProjectorElementBuildDeskriptor>obj;
+    return (
+        deskriptor.slideOptions !== undefined &&
+        deskriptor.getBasicProjectorElement !== undefined &&
+        deskriptor.getTitle !== undefined
+    );
+}
 
-    /**
-     * The projection default name for the slide
-     */
-    getProjectionDefaultName(): string;
+export interface ProjectorElementBuildDeskriptor {
+    slideOptions: SlideOptions;
+    projectionDefaultName?: string;
+    getBasicProjectorElement(): IdentifiableProjectorElement;
 
     /**
-     * The (optional) id for the slide
+     * The title to show in the projection dialog
      */
-    getIdForSlide(): number | null;
+    getTitle(): string;
+}
 
-    /**
-     * The slide's name
-     */
-    getNameForSlide(): string;
+export function isProjectable(obj: any): obj is Projectable {
+    return (<Projectable>obj).getSlide !== undefined;
```

Comment:
should maybe be `(<Projectable>obj).getSlide()`
`getSlide` would return the body of a function, that will never be undefined